interface/views.py

```diff
@@ -350,14 +349,58 @@ def code_view(request, pk, filename):
     submission = get_object_or_404(Submission, pk=pk)
     all_tas = submission.assignment.course.teaching_assistants.all()
 
-    if request.user not in all_tas:
+    if request.user not in all_tas and request.user is not submission.user:
         return HttpResponse(status=403)
 
     tree = tree_view(request, submission)
 
+    if filename == "code":
```

Comment:
This might introduce the issue of actually having a file named `code` in the archive. A solution could be having two paths in our `urlpatterns`:
```
"submission/<int:pk>/code/",
"submission/<int:pk>/code/<path:filename>/",
```
Those two should point to different views. One will handle the if case and the other the else case. It will also make it clearer where you are in your logic (empty page / page with code).